core/src/main/java/com/linecorp/armeria/client/ArrayBasedEventLoopState.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import io.netty.channel.EventLoop;
+
+final class ArrayBasedEventLoopState extends AbstractEventLoopState {
+
+    private final List<AbstractEventLoopEntry> entries = new ArrayList<>();
+    private final int maxNumEventLoops;
+    private int allActiveRequests;
+
+    ArrayBasedEventLoopState(List<EventLoop> eventLoops, int maxNumEventLoops,
+                             DefaultEventLoopScheduler scheduler) {
+        super(eventLoops, scheduler);
+        this.maxNumEventLoops = maxNumEventLoops;
+        if (eventLoops.size() == maxNumEventLoops) {
+            // We use all event loops so initialize early.
+            init(0);
+        } else {
+            init(scheduler().acquisitionStartIndex(maxNumEventLoops));
+        }
+    }
+
+    private void init(final int acquisitionStartIndex) {
+        final int initialEventLoopOffset = ThreadLocalRandom.current().nextInt(maxNumEventLoops);
+        final Function<Integer, Integer> acquisitionIndex =
+                offset -> acquisitionStartIndex + (initialEventLoopOffset + offset) % maxNumEventLoops;
+        entries.addAll(IntStream.rangeClosed(0, maxNumEventLoops - 1)
+                                .mapToObj(i -> new Entry(this, eventLoops().get(acquisitionIndex.apply(i)), i))
+                                .collect(Collectors.toList()));
+    }
```

Comment:
Init immediately, so we don't need to acquisition index helper class.
Thus remove it.